### PR TITLE
New remsql version (standalone queries only) that runs over cdb2api.

### PR DESCRIPTION
**Pull request by dorinhogea**
The latest version of remsql, available for standalone queries, runs the latter over cdb2api instead of the remsql custom protocol.  
There are a few additions to cdb2api, which are done using SET instructions (to minimize impact);
- passing protocol version (this lets live cross-cluster connections between different fdb versions)
- passing table version (required by distributed schema change versioning)
- passing cursor id (for tracking)
- passing schema morphing (specific for remote sql, simulates covering indexes for remote schemas)
Three new tunables:
- fdb_remsql_cdb2api: switches  standalone queries to cdb2api protocol
- fdb_default_version: testing only, configure a node to speak a specific remsql version  by default (a remote node can downgrade this to a lower version, for backward compatibility)
- fdb_version_emulate_precdbapi: testing only, configure a remote remsql node to fail to parse new SET options and behave like 8.0

NOTE: this relies on the cdb2api ability to pass rows in sqlite format, which was developed initially to support fdb push (forwarding full sql to remote nodes).

### CODE REVIEW

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Error &#x26A0;.<br>Regression testing: <b>10/574 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_inserts_logicalsc_generated [setup failure]<br>comdb2sys_queueodh_generated [setup failure]<br>fdb_push [setup failure]<br>biginplace [setup failure]<br>sc_resume_logicalsc_generated<br>sc_resume<br>tunables<br>auth_dba<br>truncatesc_ufid_on_generated<br>auth</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Error &#x26A0;.<br>Regression testing: <b>8/574 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>writes_remsql [setup failure]<br>misstable_remsql [setup failure]<br>basic_sslnoproto_generated [setup failure]<br>sc_resume<br>timepart_trunc_serialsc_generated<br>timepart_trunc<br>op_makerecord<br>dbcreate</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>6/575 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>timepart_retention1 [setup failure]<br>verify_error_partial_index_off_generated [setup failure]<br>basic_rowlocks_generated [setup failure]<br>basic_nogenid48_generated [setup failure]<br>analyze_exit_immediately<br>truncatesc_ufid_on_generated</p>

**Comment by dorinhogea** [comment]
db2test   Aug 26 14:02:30 2024  in-comdb2db remsqlcdb2api.R20240826.1

File: plugins/newsql/newsql.c [hunk 1]
```diff
@@ -1953,6 +1944,12 @@ int process_set_commands(struct sqlclntstate *clnt, CDB2SQLQUERY *sql_query)
                 } else {
                     rc = ii + 1;
                 }
+            } else if (strncasecmp(sqlstr, "remsql_", 7) == 0) {
+                sqlstr += 7;
+
+                clnt->remsql_set.is_remsql = 1;
+                if (process_fdb_set_cdb2api(clnt, sqlstr, err, sizeof(err))) {
```
Comment (mohitkhullar) [inline plugins/newsql/newsql.c:1951]: incorrect syntax. Remove { ?

Comment (dorinhogea) [reply plugins/newsql/newsql.c:1951]: rebase issue

File: db/fdb_fend.c [hunk 2]
```diff
@@ -5140,3 +5288,489 @@ void fdb_systable_info_free(void *data, int npoints)
...
+    start_rpc = osql_log_time();
+
+    /* remsql set otions */
+    rc = _fdb_cdb2api_send_set(fdbc);
+    if (rc)
+        return rc;
+
+    /* client set options */
+    rc = _fdb_client_set_options(pCur->clnt, hndl);
+
```
Comment (mohitkhullar) [inline db/fdb_fend.c:5469]: we can add this logic before cdb2_run_statement to make it work over IAM enabled databases
https://github.com/bloomberg/comdb2/blob/f79e2c54608b693bf67c1387cc583cc2f7c76ccf/db/fdb_push.c#L287

Comment (dorinhogea) [reply db/fdb_fend.c:5475]: done, thank you

**Review by mohitkhullar — APPROVED** [review]
lgtm

**Comment by dorinhogea** [comment]
@mohitkhullar thank you!

**Pull request merged**